### ISSUE TITLE
New dwyler invoicing app / process

### ISSUE THREAD

**Issue by rub1e** [dwyl/hq#470]
The [current invoicing app](invoice.dwyl.com) is a yuge time-saver. **Yuge**.

But there are some bugs 🐞 associated both with the dwyler-facing form, and with the add-on used to generate pdf invoices (`autocrat` 👮) :

* #468 - emails come from generic no-reply address
* #450 - invoice numbers aren't universally unique
* #423 - multiple emails sometimes sent when invoice submitted
* https://github.com/dwyl/process-handbook/issues/97 - form "leaks" personal data

For my money, these are minor to medium niggles and don't justify a redesign or new app

But if anyone disagrees or if we identify more bugs / issues, let's bump the priority up and start thinking about `version 2`

**Comment by nelsonic** [comment]
Hi @rub1e _thank you_ for opening this issue to _capture_ the "issues" with invoicing.
Invoicing was always intended to be a "feature" of our "**Time Tracking**" app; _**not** a **separate** "App"_.

So the "workflow" _should_ be:
+ **Product Owner** creates "Stories" based on feature request from People Using **Product** ("Users")
+ Stories are broken down into "Tasks" (_during Sprint Planning_) by **Dev Team**
(_Tasks should never take longer than a Day and ideally should be **one "pomodoro"**_)
+ Each Task gets _Estimated_ `before` work starts 
  + The System should not _allow_ "assignment" of a task before it has been estimated.
  + When _Estimating_ the Time required to Build the Feature, Documentation should _always_ be factored into the estimate; Our Workflow should _Systematically Eliminate_ [Tacit Knowledge](https://en.wikipedia.org/wiki/Tacit_knowledge) 
  + Features that are **Not Documented** are **_Not_** "***Done***". 
+ **Dev** assigns task to themself which tracks their time against "Task" `while` they are working.
+ **Dev** _Logs_ their progress in the Task `while` they are working so learning is captured for posterity on _everything_ that we do. 


  > _Not to "pick on" anyone in particular, but we have **way too many** instances 
of people **invoicing** us **before** their **work** is **complete** i.e. fully documented & tested_ ... 😢 
_this would **never** happen if the "workflow" was structured correctly. 
i.e. no separate "Invoicing App"._

+ At the end of `each` day, **Dev** reviews how much time they spent on a Task vs. _original_ Time Estimate - writes a _note_ in the Task indicating ***why*** something took longer than expected/predicted.
  + **_Different_ Dev** _must_ confirm "unpredicted obstacles/complexity" note has been logged on Task _before_ a time-overrun can be "confirmed", otherwise **PO** has _zero visibility_ over why Tasks take longer than expected and the whole "workflow" breaks down into _chaos_! 🔥 
+ Once Task is dev-complete it is reviewed by **QA** then, if PR approved, deployed to "Staging"
+ **PO** tests a _Feature_ (Story) on the Staging Environment and _approves_ if acceptance criteria are met.
+ When the Task is deemed "done" it is deployed _automatically_ to Production.
+ The Task is now "Awaiting Payment" and no further "action" is required from the Dev.
+ The System _automatically_:
  + Calculates how much **Time Spent** on Task(s) for the Sprint.
  + Prepares the "invoice" and sends a PDF to the **Dev** for their own records (with links to Tasks).
  + Adds the invoice to the Payment Run.
+ **Financial Controller** Approves Payment Run once per week with a _single_ click 
(_because all checks on "Time Spent" were done at a prior stage in the "workflow". 
Since Time logging has rules we cannot "over-log" so there should **never** be "invoice errors"_)

That's how it _should_ work. 😉 
The Dev should _never_ have to _manually_ enter data for an Invoice.

**Comment by iteles** [comment]
Agreed, I think we're all very aware of this. But this won't be done until 'Time' is prioritised. 

Thanks for capturing the wider issue (root cause) @nelsonic 👍
